### ISSUE TITLE
[0.76.2] Cannot reset color scheme

### ISSUE THREAD

**Issue by sangonz193** [facebook/react-native#47725]
### Description

Calling `Appearance.setColorScheme(null)` or `Appearance.setColorScheme(undefined)` no longer resets the color scheme `useColorScheme` returns like on previous rn versions.

### Steps to reproduce

1. Call `Appearance.setColorScheme(null)` or `Appearance.setColorScheme(undefined)`
2. Next call to `useColorScheme()` will return `null` or `undefined`, instead of `"light"` or `"dark"` based on the OS setting.

### React Native Version

0.76.2

### Affected Platforms

Runtime - Android, Runtime - iOS

### Output of `npx react-native info`

```text
System:
  OS: macOS 15.1
  CPU: (10) arm64 Apple M1 Pro
  Memory: 130.08 MB / 16.00 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 22.11.0
    path: ~/.nvm/versions/node/v22.11.0/bin/node
  Yarn:
    version: 1.22.22
    path: ~/.nvm/versions/node/v22.11.0/bin/yarn
  npm:
    version: 10.9.0
    path: ~/.nvm/versions/node/v22.11.0/bin/npm
  Watchman:
    version: 2024.11.11.00
    path: /opt/homebrew/bin/watchman
Managers:
  CocoaPods:
    version: 1.16.2
    path: /opt/homebrew/bin/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 24.1
      - iOS 18.1
      - macOS 15.1
      - tvOS 18.1
      - visionOS 2.1
      - watchOS 11.1
  Android SDK: Not Found
IDEs:
  Android Studio: 2024.2 AI-242.23339.11.2421.12550806
  Xcode:
    version: 16.1/16B40
    path: /usr/bin/xcodebuild
Languages:
  Java:
    version: 17.0.13
    path: /usr/bin/javac
  Ruby:
    version: 2.6.10
    path: /usr/bin/ruby
npmPackages:
  "@react-native-community/cli":
    installed: 15.1.2
    wanted: ^15.1.2
  react:
    installed: 18.3.1
    wanted: 18.3.1
  react-native:
    installed: 0.76.2
    wanted: 0.76.2
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: Not found
  newArchEnabled: Not found
iOS:
  hermesEnabled: true
  newArchEnabled: true
```

### Stacktrace or Logs

```text
No error message
```

### Reproducer

https://snack.expo.dev/@sangonz193/frowning-green-donut

### Screenshots and Videos

Expo 52 shows the bug, Expo 51 shows how it behaved before. Device used here was configured for dark mode.

https://github.com/user-attachments/assets/59671cda-033c-46ce-914c-9e9aea3629d1

https://github.com/user-attachments/assets/7fc2645f-9d8b-4714-900c-fa0915686bfb

**Comment by sangonz193** [comment]
Patching `node_modules/react-native/Libraries/Utilities/Appearance.js` seems to be working:
```diff
   const {NativeAppearance} = state;
   if (NativeAppearance != null) {
     NativeAppearance.setColorScheme(colorScheme ?? 'unspecified');
-    state.appearance = {colorScheme};
+    state.appearance = {colorScheme:getColorScheme()};
   }
```

Unsure whether this would be the desired fix.

**Comment by enchorb** [comment]
why is this going into 0.77 rather than a patch release for 0.76. This is broken atm.

**Comment by SMYusupov** [comment]
Agree with @enchorb. @cipolleschi as this is a bug, could it be also merged to 0.76 version?

**Comment by cortinico** [comment]
> could it be also merged to 0.76 version?

You can request a pick request for this over here:
https://github.com/reactwg/react-native-releases/#pick-requests-and-support-policy

**Comment by SMYusupov** [comment]
> > could it be also merged to 0.76 version?
> 
> You can request a pick request for this over here: https://github.com/reactwg/react-native-releases/#pick-requests-and-support-policy

Requested one, thanks @cortinico 

Link to request - https://github.com/reactwg/react-native-releases/issues/747
